### PR TITLE
feat: theme overrides for Card, CardContent, CardActions

### PR DESCRIPTION
**Pull request by seigler**
image

Simple Card component supporting a heading image.

Also in this PR:
* pre-commit and pre-push hooks now only lint affected code
* Material UI shadows in the Prenda theme have been lightened 50%

### CODE REVIEW

**Comment by seigler** [comment]
Please merge #70 before this so the release tags are in the right order

File: libs/spark/src/theme/theme.ts
Comment (shadoath) [inline libs/spark/src/theme/theme.ts:136]: Can you explain this to me?

Comment (seigler) [reply libs/spark/src/theme/theme.ts:136]: This is the same code Google uses to generate the different levels of shadow corresponding to the `elevation` prop on most of their components. But I changed the opacity constants in `shadows.ts` to be 50% lighter to more closely match Spencer's designs.

File: libs/spark/src/Card.tsx
Comment (shadoath) [inline libs/spark/src/Card.tsx:1]: Can be removed.

File: libs/spark/src/theme/theme.ts
Comment (seigler) [reply libs/spark/src/theme/theme.ts:136]: The numbers correspond to pixel sizes for shadow location, offset, and blur.

**Review by shadoath — COMMENTED** [review]
Minor change. 
Didn't know about https://placekitten.com fun service for sure.

File: libs/spark/src/theme/theme.ts
Comment (shadoath) [reply libs/spark/src/theme/theme.ts:136]: Ahhh, then the array index correlates to the `elevation` wanted.

**Review by WilliamKelley — CHANGES_REQUESTED** [review]
This PR is missing following the [Contributor](https://github.com/prenda-school/prenda-spark/blob/main/CONTRIBUTING.md#new) guideline on new components and on implementation work. Doing so could alleviate a lot of my review. The customization of theme shadows and the correcting of the lint command should have been separate PR's. Those could have been approved/merged no problem, without being held up by Card review.

(Note: I don't mean to be over-bearing / pedantic here; I push because I think each of the guidelines has value. If it's ill formed / too strict, lets work together to improve it.)

This implementation could be greatly simplified. This biggest issue is forcing the first child of `Card` to be `CardContent`, which is unnecessary / harmful to use-cases. MUI has [several components related to Cards](https://material-ui.com/components/cards/#api). Spark would be better to customize each of these individually, and leave assembly up to the consumer -- this is standard re-usability under React. Spark will undoubtedly need Cards with varying layouts (text, image organizations, and other) in the future, and assuming one composition will force a rewrite / breaking change in the future.

Action items
- Adhere to guidelines
- Customize one component per file
- Customize each card component needed

File: libs/spark/src/theme/theme.ts
Comment (WilliamKelley) [reply libs/spark/src/theme/theme.ts:136]: This is very nice to have customized!

Also small note: Material-UI /= Google. The connection between them is that Material-UI creates libs that implement Google's published 'Material Design', but they're very much different companies.

**Comment by WilliamKelley** [comment]
I don't think this PR should version itself, it poses problems for changelog generation and review, see #72

File: libs/spark/src/Card.tsx
Comment (seigler) [reply libs/spark/src/Card.tsx]: Thanks for this comment. In this instance it has become moot

File: libs/spark/src/theme/shadows.ts
Comment (seigler) [reply libs/spark/src/theme/shadows.ts]: This was an excellent suggestion, which I've included.

File: libs/spark/stories/card.stories.tsx
Comment (seigler) [reply libs/spark/stories/card.stories.tsx]: Typography needs its own theme PR to match Spencer's designs.

**Comment by seigler** [comment]
I rolled back the versioning and release tag, and moved the changes to Card and CardContent defaults into the theme.

File: libs/spark/stories/card.stories.tsx
Comment (WilliamKelley) [reply libs/spark/stories/card.stories.tsx]: we have a typography component :)

File: libs/spark/stories/card.stories.tsx
Comment (WilliamKelley) [inline libs/spark/stories/card.stories.tsx:35]: let's one-off customize this and add a `max-width: 580px` so it's not huge in the story. I chose that value from the figma page example.

File: libs/spark/stories/card.stories.tsx
Comment (WilliamKelley) [inline libs/spark/stories/card.stories.tsx:21]: For each of these stories, I'd suggest copying the text/content in the figma pages. It doesn't matter technically, but just to validate the overall look with a bunch of text is nice.

**Review by WilliamKelley — COMMENTED** [review]
Overrides are so nice. The stories can advantage Spark's Typography.

There are a few other elements showcased in the Card Figma page: 'action' Button in the Footer, Icon Button at the top right, Tag beside title. We don't have tags yet, but we have the other two. Do you want to add validation and stories for them in this PR or leave it as is?

File: libs/spark/stories/card.stories.tsx
Comment (seigler) [reply libs/spark/stories/card.stories.tsx]: :man_facepalming: yep, there it is. Nice

File: libs/spark/stories/card.stories.tsx
Comment (seigler) [reply libs/spark/stories/card.stories.tsx]: I like this suggestion.

File: libs/spark/stories/card.stories.tsx
Comment (seigler) [reply libs/spark/stories/card.stories.tsx]: I updated both Card examples to match their Figma counterparts.

File: libs/spark/stories/card.stories.tsx
Comment (WilliamKelley) [inline libs/spark/stories/card.stories.tsx:22]: Match figma
```suggestion
      <Typography variant="paragraph-xl" color="textOnLight">
```

File: libs/spark/stories/card.stories.tsx
Comment (WilliamKelley) [inline libs/spark/stories/card.stories.tsx:65]: Match figma
```suggestion
      <Typography variant="paragraph-xl" color="textOnLight">
```

File: libs/spark/stories/card.stories.tsx
Comment (WilliamKelley) [inline libs/spark/stories/card.stories.tsx:62]: Same as above
```suggestion
      <Typography variant="heading-md" color="textOnLight" style={{ marginBottom: '1rem' }}>
```

**Review by WilliamKelley — APPROVED** [review]
Small changes to exactly line up with Figma. Good to merge after they're addressed.

_Edit: actually, before merge, I'll go in and accept the new stories as "baseline" in Chromatic_

File: libs/spark/src/CardContent.tsx
Comment (WilliamKelley) [inline libs/spark/src/CardContent.tsx:3]: Can you check if this works & include if you think it's a good idea. Finding the right paradigm for padding when the children can vary is always hard.
```suggestion
    padding: '24px 24px 0',
    '&:last-child': {
      paddingBottom: 24;
    }
```

Comment (seigler) [reply libs/spark/src/CardContent.tsx:3]: The last-child rule is already present from the Mui default styles, so this combination of rules already applies to MuiCardContent.

File: libs/spark/stories/card.stories.tsx
Comment (seigler) [reply libs/spark/stories/card.stories.tsx:62]: Might it be better to change the `marginBottom` rule for `Typography[gutter]`?

File: libs/spark/src/CardContent.tsx
Comment (WilliamKelley) [reply libs/spark/src/CardContent.tsx:3]: Ah okay, great

File: libs/spark/stories/card.stories.tsx
Comment (WilliamKelley) [reply libs/spark/stories/card.stories.tsx:62]: Possibly, it's a bit of a toss up though because a 1rem gutter is desirable on a heading, but probably not on a paragraph or smaller text...

Comment (WilliamKelley) [reply libs/spark/stories/card.stories.tsx:62]: Let's change it one-off for now but keep it in mind. If this keeps popping up, we can set a better default.

Comment (WilliamKelley) [reply libs/spark/stories/card.stories.tsx:62]: Looking at how Material-UI does it for their docs, they set a top/bottom margin for header variants and avoid `gutterBottom` on them. Spark doesn't have such a specification, but filing a design request would be the way to go for that.

**Comment by WilliamKelley** [comment]
Great work. I've accepted the stories as baselines in Chromatic, and changed, the title to a Conventional Commit.

I'll squash and merge this time to see if what I'm proposing in #72 works, and post as an example.

**Pull request merged**
